go_covert_lib/controller/processor/symmetricEncryption/symmetricEncryption_config.go

```diff
@@ -0,0 +1,40 @@
+package symmetricEncryption
+
+import (
+	"../../config"
+)
+
+type ConfigClient struct {
+	Algorithm config.SelectParam
+	Mode       config.SelectParam
+	Key        config.HexKeyParam
+}
+
+func GetDefault() ConfigClient {
+	return ConfigClient{
+		Algorithm: config.MakeSelect("Advanced Encryption Standard (AES)", 
+					[]string{"Advanced Encryption Standard (AES)", "Data 
+					Encryption Standard (DES)", "Triple Data Encryption 
+					Standard (3DES)"}, config.Display{Description: 
+					"Select an encryption algorithm"}),
+		Mode:       config.MakeSelect("Galois Counter Mode (GCM)", 
+					[]string{"Galois Counter Mode (GCM)", 
+					"Cipher Block Chaining (CBC)", "Cipher Feedback (CFB)", 
+					"Counter (CTR)"}, config.Display{Description: 
+					"Select the mode of operation"}),
+		// AES-128 = key size 32 characters long, AES-192 = key size 48 
+		//characters long, and AES-256 = key size 64 characters long
+		// DES key size 16 characters long, 3DES key size 48 characters
+		// long (i.e. 3*16)
+		// NOTE: key byte is 2 characters hence the options are 8, 16, 24, or 32
+		Key: config.MakeHexKey(make([]byte, 32), []int{8, 16, 24, 32}, 
+			 config.Display{Description: "The shared secret key used for 
+			 Advanced Encryption Standard (AES) must be 32, 48 or 64 
+			 characters in length, for Data Encryption Standard (DES) 
+			 must be 16 characters in length, and for Triple Data Encryption
+			  Standard (3DES) must be 48 characters in length"})}
+}
+
+func ToProcessor(aes ConfigClient) (*SymmetricEncryption, error) {
```

Comment:
You should add code here to check that the Algorithm and Mode are valid values from the list. If you did that you could run all the "block, err = des.NewTripleDESCipher(c.key)" lines here instead of each time the processor runs, saving you runtime during processing. It would also mean that the key would be validated when the key is set instead of when the user tries to write a message, which is better for the user.